cacheops/redis.py

```diff
@@ -22,12 +25,51 @@ def handle_connection_failure(call):
             warnings.warn("The cacheops cache timed out! Error: %s" % e, RuntimeWarning)
         except redis.RedisError as e:
             logger.exception(e)
+
+    if settings.CACHEOPS_DEGRADE_PERSISTENT_PER_REQUEST:
+        degraded_client_set = set()
+
+
+        class DegradedClientError(redis.RedisError):
+            """ Error which raised after attempt to make a query to the degraded
+            client.
+            """
+
+
+        @decorator
+        def degrade_client_decorator(call):
+            """ Decorator which marks client as degraded if Connection or Timeout
+            error occured during the query.
+            Query to the degraded client won't be executed at all.
+            Degraded mark saves during the request.
+            """
+            if call.self in degraded_client_set:
+                raise DegradedClientError()
+
+            try:
+                return call()
+            except (redis.ConnectionError, redis.TimeoutError):
```

Comment:
I looked again on the redis exceptions and decided that more safe way is to catch only connection and timeout errors. So nothing to change there.